code/modules/power/singularity/singularity.dm

```diff
@@ -350,6 +365,31 @@
 					return TRUE
 	. = !.
 
+/obj/singularity/proc/pick_random_target()
+	var/list/sections = list()
+	for (var/section_x = -SINGULARITY_QUADRANT_DISTANCE - SINGULARITY_QUADRANT_SIZE; section_x <= SINGULARITY_QUADRANT_DISTANCE - SINGULARITY_QUADRANT_SIZE; section_x += SINGULARITY_QUADRANT_SIZE)
+		for (var/section_y = -SINGULARITY_QUADRANT_DISTANCE - SINGULARITY_QUADRANT_SIZE; section_y <= SINGULARITY_QUADRANT_DISTANCE - SINGULARITY_QUADRANT_SIZE; section_y += SINGULARITY_QUADRANT_SIZE)
+			var/turf/section_loc = locate(x + section_x, y + section_y, z)
+			if (!section_loc || !istype(section_loc))
+				continue
+			var/list/box = get_box(x + section_x, y + section_y, z, SINGULARITY_QUADRANT_SIZE, SINGULARITY_QUADRANT_SIZE)
+			var/interest = 0
+			for (var/turf/T in box)
+				if (!isspaceturf(T))
+					interest += 2
+				var/list/objs = list()
+				for (var/A in T.contents)
+					if (istype(A, /atom/movable))
+						objs += A
+				interest += CEILING(length(objs) / 7.5, 0.5)
```

Comment:
Since you don't actually do anything with the objects, perhaps it'd be better to just count them rather than put them in a list and *then* count them.

I doubt it's a huge performance difference but it'd be a nice change.